src/components/course/data/hooks.js

```diff
@@ -168,33 +169,59 @@ export function useCoursePriceForUserSubsidy(activeCourseRun, userSubsidy) {
         return null;
       }
 
-      const priceDetails = {
-        list: numberWithPrecision(listPrice),
+      const onlyListPrice = {
+        list: listPrice,
       };
-      if (!userSubsidy) {
-        return priceDetails;
-      }
 
-      const { discountType, discountValue } = userSubsidy;
-      let discountedPrice;
+      // Case 1: License Subsidy
+      if (userSubsidy && userSubsidy.discountValue) {
+        const { discountType, discountValue } = userSubsidy;
+        let discountedPrice;
+
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
+          discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        }
+
+        if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
+          discountedPrice = Math.max(listPrice - discountValue, 0);
+        }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.PERCENTAGE) {
-        discountedPrice = listPrice - (listPrice * (discountValue / 100));
+        if (isDefinedAndNotNull(discountedPrice)) {
+          return {
+            ...onlyListPrice,
+            discounted: discountedPrice,
+          };
+        }
+        return {
+          ...onlyListPrice,
+          discounted: onlyListPrice.list,
+        };
       }
 
-      if (discountType === SUBSIDY_DISCOUNT_TYPE_MAP.ABSOLUTE) {
-        discountedPrice = Math.max(listPrice - discountValue, 0);
+      // Case 2: No subsidy, no offers for course
+      if (!offers) {
+        return onlyListPrice;
       }
 
-      if (isDefinedAndNotNull(discountedPrice)) {
-        priceDetails.discounted = numberWithPrecision(discountedPrice);
-      } else {
-        priceDetails.discounted = priceDetails.list;
+      const offerForCourse = findOfferForCourse(offers, catalogList);
+
+      if (!offerForCourse) {
+        return onlyListPrice;
       }
 
-      return priceDetails;
+      // Case 3: No subsidy, but found offer for course
+      const { benefitValue, usageType } = offerForCourse;
+      if (usageType === 'Percentage') {
+        // only 100% discount supported as of now
+        const discountedPrice = listPrice - (listPrice * (benefitValue / 100));
+        return {
+          ...onlyListPrice,
+          discounted: discountedPrice,
+        };
+      }
+      return onlyListPrice;
```

Comment:
This logic feels unnecessary duplicative from the logic for Case 1 above. See my larger discussion comment above about the original intentions of the `userSubsidy` variable. Largely, I believe we can simplify this logic by using a generic interface that doesn't differentiate between subsidy types.

Reply:
aha I like that already. I think subsidy is just a subsidy right? I will take a look at this. The interface idea is superb and we should be able to simplify this

Reply:
Correct, a subsidy can be considered anything an Enterprise Customer provided to their employee for a discounted price (partial/full).

So, this includes subscription licenses, codes, and Enterprise Offers today.

Reply:
Yeah I do miss working with java, in cases like this. there the interface idea is enforced! we can do it with js (ts) too but it's extra lift.. anyway I think the type of discount should be used somewhere just not in this piece of code right? I figure we still need to. know offer vs license vs etc in a layer above this but the interface is the same for each case., refactoring this piece now

Reply:
also makes me think the proptypes for this needs to be updated to catch this

Reply:
> I figure we still need to. know offer vs license vs etc in a layer above this but the interface is the same for each case.

My thoughts exactly :) 

This is where the existing `userSubsidy` variable is initialized: https://github.com/edx/frontend-app-learner-portal-enterprise/blob/e1e4c7a90c9146e6d6aa441cd919e916d2d6db40/src/components/course/data/service.js#L43

It relies on making the call to the `license-subsidy` endpoint. In the case of offers (codes), we already know all the codes data from the `UserSubsidyContext`, we just don't know which code applies to the catalog for which this course is part of.

I think we could add some business logic here to decide which subsidy type best applies to the specific course:
* If the user's subscription license is valid for this course, `userSubsidy` becomes the license data.
* If the user's subscription license is _not_ valid for the course (or the user does not have a license at all), and user has a code that applies to the specific course, `userSubsidy` becomes the code data.
* If no subsidies exist for the user, or no subsidy can be applied to the course (i.e., the license/code applies to a different catalog not associated with the subsidy), `userSubsidy` becomes `null` and implies the user would be paying full price out-of-pocket.

Reply:
this makes sense. I think the userCoursePriceForSubsidy hook should be free of logic about known type of subsidy I will make that change thanks!